src/SqsPoller/SqsPollerHostedService.cs

```diff
@@ -79,17 +80,25 @@ public async Task HandleMessage(Message message, CancellationToken cancellationT
                 _logger.LogTrace("Message {message_id} {receipt_handle} has been handled");
 
                 _logger.LogTrace("Deleting the message with id {message_id} and ReceiptHandle {receipt_handle}");
-                _amazonSqsClient.DeleteMessageAsync(
-                    new DeleteMessageRequest
+                var deleteRequest = new DeleteMessageRequest
+                {
+                    QueueUrl = queueUrl,
+                    ReceiptHandle = message.ReceiptHandle
+                };
+                _amazonSqsClient.DeleteMessageAsync(deleteRequest, cancellationToken)
+                    .ContinueWith(deleteMessageTask =>
                     {
-                        QueueUrl = queueUrl,
-                        ReceiptHandle = message.ReceiptHandle
+                        _logger.LogTrace("Message {message_id} {receipt_handle} has been deleted");
+                        if (deleteMessageTask.IsFaulted)
+                        {
+                            _config.OnException?.Invoke(deleteMessageTask.Exception,
```

Comment:
I don't think that one exception handler `config.OnException` is a good idea to different kind of exceptions(
delete  a message faulted, a message is not handled, messages are not read from a queue). I suggest implementing `OnException` per type or passing ExceptionType enum in parameters instead of message.

Reply:
@AlexeyBogdan95 Updated, please check & approve PR